global_helpers/panther_ipinfo_helpers.py

```diff
@@ -0,0 +1,89 @@
+from panther_base_helpers import deep_get
+
+
+class PantherIPInfoException(Exception):
+    def __init__(self):
+        message = "Please enable both IPInfo Location and ASN Lookup Tables"
+        super().__init__(message)
+
+
+class IPInfoLocation:
+    def __init__(self, event):
+        self.ipinfo_location = deep_get(event, "p_enrichment", "ip-info-location-cidr")
+
+    def city(self, match_field) -> str:
+        return deep_get(self.ipinfo_location, match_field, "city")
+
+    def country(self, match_field) -> str:
+        return deep_get(self.ipinfo_location, match_field, "country")
+
+    def latitude(self, match_field) -> str:
+        return deep_get(self.ipinfo_location, match_field, "lat")
+
+    def longitude(self, match_field) -> str:
+        return deep_get(self.ipinfo_location, match_field, "lng")
+
+    def postal_code(self, match_field) -> str:
+        return deep_get(self.ipinfo_location, match_field, "postal_code")
+
+    def region(self, match_field) -> str:
+        return deep_get(self.ipinfo_location, match_field, "region")
+
+    def region_code(self, match_field) -> str:
+        return deep_get(self.ipinfo_location, match_field, "region_code")
+
+    def timezone(self, match_field) -> str:
+        return deep_get(self.ipinfo_location, match_field, "timezone")
+
+
+class IPInfoASN:
+    def __init__(self, event):
+        self.ipinfo_asn = deep_get(event, "p_enrichment", "ip-info-asn-cidr")
+
+    def asn(self, match_field) -> str:
+        return deep_get(self.ipinfo_asn, match_field, "asn")
+
+    def domain(self, match_field) -> str:
+        return deep_get(self.ipinfo_asn, match_field, "domain")
+
+    def name(self, match_field) -> str:
+        return deep_get(self.ipinfo_asn, match_field, "name")
+
+    def route(self, match_field) -> str:
+        return deep_get(self.ipinfo_asn, match_field, "route")
+
+    def asn_type(self, match_field) -> str:
+        return deep_get(self.ipinfo_asn, match_field, "type")
+
+
+def get_ipinfo_location_object(event):
+    if deep_get(event, "p_enrichment", "ip-info-location-cidr"):
+        return IPInfoLocation(event)
+    return None
+
+
+def get_ipinfo_asn_object(event):
+    if deep_get(event, "p_enrichment", "ip-info-asn-cidr"):
+        return IPInfoASN(event)
+    return None
+
+
+def geoinfo_from_ip(event, match_field):
```

Comment:
@rleighton What should I do about the two fields that are not present in IPInfo's data? (hostname and anycast)